scripts/run_trade_bot.sh

```diff
@@ -132,28 +132,99 @@ do
 
   for (( pair_index=0; pair_index<$token_pair_config_array_length; pair_index++ ))
   do
-    token_pair=$( echo "$token_pair_config_array" | jq -r ".[$pair_index].pair | sort_by(.denom)" )
+    token_pair=$( echo "$token_pair_config_array" | jq -r ".[$pair_index].pair" )
     token_pair_config=$( echo "$token_pair_config_array" | jq -r ".[$pair_index].config" )
 
-    # pair simulation options
+    # common pair config variables
     token0=$( echo "$token_pair" | jq -r '.[0].denom' )
     token1=$( echo "$token_pair" | jq -r '.[1].denom' )
     token0_total_amount=$( echo "$token_pair" | jq -r '.[0].amount' )
     token1_total_amount=$( echo "$token_pair" | jq -r '.[1].amount' )
     tick_count=$( echo "$token_pair_config" | jq -r '.ticks' )
     tick_count_on_each_side=$(( $tick_count / 2 ))
-    # convert price to price index here
-    price_index=$( echo "$token_pair_config" | jq -r '((.price | log)/(1.0001 | log) | round)' )
     fees=$( echo "$token_pair_config" | jq -r '.fees' )
     rebalance_factor=$( echo "$token_pair_config" | jq -r '.rebalance_factor' )
     deposit_factor=$( echo "$token_pair_config" | jq -r '.deposit_factor' )
     swap_factor=$( echo "$token_pair_config" | jq -r '.swap_factor' )
     deposit_index_accuracy=$( echo "$token_pair_config" | jq -r '.deposit_accuracy' )
     swap_index_accuracy=$( echo "$token_pair_config" | jq -r '.swap_accuracy' )
-    amplitude1=$( echo "$token_pair_config" | jq -r '.amplitude1' )
-    amplitude2=$( echo "$token_pair_config" | jq -r '.amplitude2' )
-    period1=$( echo "$token_pair_config" | jq -r '.period1' )
-    period2=$( echo "$token_pair_config" | jq -r '.period2' )
+    price_config=$( echo "$token_pair_config" | jq -r '.price' )
+
+    # if price is a number, i.e. if price is set manually
+    if (( $(echo "$price_config" | grep -c '^[0-9]\+\(\.[0-9]\+\)\?$') == 1 ))
+    then
+      echo "crafting $token0<>$token1 price using config approximation params" > /dev/stderr
+
+      # pair simulation options
+      amplitude1=$( echo "$token_pair_config" | jq -r '.amplitude1' )
+      amplitude2=$( echo "$token_pair_config" | jq -r '.amplitude2' )
+      period1=$( echo "$token_pair_config" | jq -r '.period1' )
+      period2=$( echo "$token_pair_config" | jq -r '.period2' )
+
+      # convert price to price index here
+      price_index=$( echo "$token_pair_config" | jq -r '((.price | log)/(1.0001 | log) | round)' )
+      echo "calculated price index before approximation $price_index" > /dev/stderr
```

Comment:
echoing to `> /dev/stderr` is not necessary in the main file, its only needed within called functions